bot/cogs/fun.py

```diff
@@ -1,257 +1,257 @@
-            value=f"https://explainxkcd.com/{number}")
-
-        await ctx.send(embed=comic)
-
-    @command()
-    async def quotes(self, ctx: Context, member: Member = None):
-        """
-        Returns a random quotation from the #quotes channel.
-        A user can be specified to return a random quotation from that user.
-        """
-        quote_channel = self.bot.get_channel(QUOTES_CHANNEL_ID)
-        quotes = self.bot.quotes
-
-        if member is None:
-            message_id = choice(quotes[choice(list(quotes.keys()))])
-        else:
-            user_quotes = quotes[f'{member.name}#{member.discriminator}']
-            if not user_quotes:
-                await ctx.send("No quotes from that user.")
-                return
-            message_id = choice(user_quotes)
-
-        message = await quote_channel.get_message(message_id)
-        await ctx.send(embed=message.embeds[0])
-
-    @command()
-    @has_any_role(*ADMIN_ROLES)
-    async def set_quote_channel(self, ctx: Context, channel: TextChannelConverter):
-        """
-        Sets the quotes channel.
-        """
-        self.quote_channel = channel
-        await ctx.send("Quotes channel successfully set.")
-
-    async def create_text_image(self, ctx: Context, person: str, text: str):
-        """
-        Creates an image of a given person with the specified text.
-        """
-        lines = textwrap.wrap(text, 15)
-        draw = Drawing()
-        image = Image(filename=f"bot/resources/{person}SaysBlank.png")
-        draw.font = "bot/resources/Dosis-SemiBold.ttf"
-        draw.text_alignment = "center"
-        draw.font_size = 34
-        offset = 45 - 10 * len(lines)
-        for line in lines:
-            draw.text(image.width // 5 + 20, image.height // 5 + offset, line)
-            offset += 35
-        draw(image)
-        image.save(filename=f"bot/resources/{person}Says.png")
-        await ctx.send(file=File(f"bot/resources/{person}Says.png"))
-
-    @command()
-    async def agentj(self, ctx: Context, *, text: str):
-        """
-        Creates an image of Agent J with the specified text.
-        """
-        await self.create_text_image(ctx, "AgentJ", text)
-
-    @command()
-    async def jibhat(self, ctx: Context, *, text: str):
-        """
-        Creates an image of Jibhat with the specified text.
-        """
-        await self.create_text_image(ctx, "Jibhat", text)
-
-
-def setup(bot):
-    """
-    Required boilerplate for adding functionality of cog to bot.
-    """
-    bot.add_cog(Fun(bot))
+"""
+Set of bot commands designed for general leisure.
+"""
+import textwrap
+from itertools import cycle
+from random import choice, randint
+from string import ascii_lowercase
+from typing import AsyncGenerator
+from urllib.parse import urlencode
+
+from aiohttp import ClientSession
+from discord import Embed, File, Message
+from discord.ext.commands import (
+    Bot, Context, TextChannelConverter, command, has_any_role, UserConverter

```

Comment:
We put the uppercase things first, then the lowercase things, and each is sorted alphabetically. In this case,
```suggestion
    Bot, Context, TextChannelConverter, UserConverter, command, has_any_role
```